tmt/steps/prepare/__init__.py

```diff
@@ -102,26 +118,28 @@ def wake(self):
             self.status('todo')
             self.save()
 
-    def show(self):
+    def show(self) -> None:
         """ Show discover details """
         for data in self.data:
+            if not isinstance(data, tmt.steps.StepData):
+                raise GeneralError('data should be an instance of StepData')
```

Comment:
This should not be needed, `self.data` is expected to be `List[StepData]`

Reply:
Right, removed